### ISSUE TITLE
Macbook air m1 wifi issue 

### ISSUE THREAD

**Issue by phvlkn** [bettercap/bettercap#916]
- Bettercap version: 2.31.1
- macOS 11.5 (same problem on 11.1) arm64 / MacBook Air M1 8/256
- go ver 1.16.3
- sudo bettercap -iface en0 (same problem wo/args)
- debug: emin@MBAemin ~ % sudo bettercap -debug -iface en0 
Password:
bettercap v2.31.1 (built for darwin arm64 with go1.16.3) [type 'help' for a list of commands]

```
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [sys.log] [dbg] arp.spoof arp cache restoration after spoofing enabled
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [sys.log] [inf] gateway monitor started ...
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [session.started] {session.started 2021-09-17 23:59:46.63123 +0400 +04 m=+0.020521293 <nil>}
192.168.1.0/24 > 192.168.1.92  » [23:59:46] [mod.started] events.stream
192.168.1.0/24 > 192.168.1.92  » wifi.reco[23:59:51] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » wifi.reco[23:59:51] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » wifi.recon on
[23:59:56] [sys.log] [inf] wifi using interface en0 (18:3e:ef:db:fa:fd)
[23:59:56] [sys.log] [dbg] wifi interface en0 txpower set to 30
[23:59:56] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
[23:59:56] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
[23:59:58] [sys.log] [dbg] wifi new frequencies: [2412 2417 2422 2427 2432 2437 2442 2447 2452 2457 2462 2467 2472 5180 5200 5220 5240 5260 5280 5300 5320 5660 5680 5700 5720 5745 5765 5785 5805 5825]
[23:59:58] [sys.log] [dbg] wifi wifi supported frequencies: [2412 2417 2422 2427 2432 2437 2442 2447 2452 2457 2462 2467 2472 5180 5200 5220 5240 5260 5280 5300 5320 5660 5680 5700 5720 5745 5765 5785 5805 5825]
[23:59:58] [sys.log] [inf] wifi started (min rssi: -200 dBm)
192.168.1.0/24 > 192.168.1.92  » [23:59:58] [mod.started] wifi
192.168.1.0/24 > 192.168.1.92  » [23:59:58] [sys.log] [inf] wifi channel hopper started.
192.168.1.0/24 > 192.168.1.92  » [23:59:58] [sys.log] [dbg] wifi wifi stations pruner started (ap.ttl:5m0s sta.ttl:5m0s).
192.168.1.0/24 > 192.168.1.92  » [00:00:01] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:01] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » wifi.[00:00:06] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » wifi.[00:00:06] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » wifi.show

en0 (ch. 140) / ↑ 0 B / ↓ 58 kB / 297 pkts

192.168.1.0/24 > 192.168.1.92  » [00:00:11] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:11] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » [00:00:16] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:16] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » [00:00:21] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:21] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » [00:00:26] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:26] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » [00:00:31] [sys.log] [dbg] gateway is 192.168.1.1[8:c6:b3:c3:2f:89]
192.168.1.0/24 > 192.168.1.92  » [00:00:31] [sys.log] [dbg] [gw] ipv4=192.168.1.1(08:c6:b3:c3:2f:89) ipv6=()
192.168.1.0/24 > 192.168.1.92  » 
```

- steps:
    1) wifi.recon on
    2) wifi.show
- expected: list of wifi actions, clients, APs 
- actually: only num of packets and data amount
- if i'm not mistaken, it was working on Intel MBP (Catalina)

**Comment by SMarioMan** [comment]
You must make sure you aren't connected to any networks for the list to populate. Otherwise, the interface will fail to enable monitor mode.

**Comment by sudo-nano** [comment]
On M1 Macs, you have to manually disconnect from your current WiFi network before monitor mode can be enabled. You can tell that you're still connected to a network and not in monitor mode if your bettercap prompt has your subnet and local IP, in this case `192.168.1.0/24 > 192.168.1.92  » `

When you disconnect from the network and restart bettercap, the prompt will list your interface, in this case `en0`, and the WiFi icon in the toolbar will now have an eye. I was able to reproduce the behavior you described, but was also able to properly detect 2.4 GHz networks when I started bettercap after disconnecting from WiFi. Unfortunately, 5 GHz network detection seems to still be broken on M1 Macs, which I will open a separate issue for.

**Comment by wzulfikar** [comment]
Thanks for the suggestion, @sudo-nano! It works for me (Macbook Air M1).

```sh
~ bettercap --version
bettercap v2.32.0 (built for darwin arm64 with go1.19.2)
```

Here's a 30-second preview of the issue. Posting here for posterity:

https://user-images.githubusercontent.com/7823011/204108603-c821dd71-3c3e-4bc0-8635-ced4a13d7f2e.mp4

**Comment by kantum** [comment]
I'm having a similar issue, everything seams to work fine but `wifi.show` only gives `en0 (ch. 1) / ↑ 0 B / ↓ 13 MB / 40383 pkts`

@wzulfikar Your video does not show the `wifi.show` command working, did you managed to make it work?

**Comment by kantum** [comment]
Duplicate #998

**Comment by Roomasheartbreak** [comment]
@kantum I'm having the same exact issue on my MacBook m2 as well, did you find a solution?

**Comment by kantum** [comment]
Nop, I gave up 🤷

**Comment by 1r0** [comment]
> I'm having a similar issue with a Macbook m2, everything seams to work fine but `wifi.show` only gives `en0 (ch. 1) / ↑ 0 B / ↓ 13 MB / 40383 pkts`
> 
> @wzulfikar Your video does not show the `wifi.show` command working, did you managed to make it work?

Yes, wifi.show list only gets populated after you select a specific channel using

`wifi.recon.channel [1-16]`

**Comment by evilsocket** [comment]
Fixed by 2.33.0
